From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Wed, 2 Dec 2020 17:05:01 +0100
Subject: fpi-device: Return proper type on identification success

Identify function is supposed to propagate a boolean value, but we make
it return an integer instead on idle, this can be normally the same in
most of architectures, but not in BE ones.

So, make it return the proper type.
Fixes test failures in s390x.

Related to #236

Origin: https://gitlab.freedesktop.org/libfprint/libfprint/-/commit/2783ac3
---
 libfprint/fpi-device.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libfprint/fpi-device.c b/libfprint/fpi-device.c
index 981df87..0968c0c 100644
--- a/libfprint/fpi-device.c
+++ b/libfprint/fpi-device.c
@@ -1157,7 +1157,7 @@ fpi_device_identify_complete (FpDevice *device,
         }
       else
         {
-          fpi_device_return_task_in_idle (device, FP_DEVICE_TASK_RETURN_INT, GINT_TO_POINTER (TRUE));
+          fpi_device_return_task_in_idle (device, FP_DEVICE_TASK_RETURN_BOOL, GUINT_TO_POINTER (TRUE));
         }
     }
   else
